runtime/nvqir/CircuitSimulator.h

```diff
@@ -8,46 +8,211 @@
 
 #pragma once
 
+#include "Gates.h"
 #include "Logger.h"
 #include "MeasureCounts.h"
 #include "NoiseModel.h"
 #include "QIRTypes.h"
+
 #include <cstdarg>
 #include <cstddef>
+#include <queue>
+#include <set>
```

Comment:
```suggestion
```
Suppose we don't need it anymore

Reply:
Yea I will remove. I also left some commented code in here, will circle back after I'm done with another branch I'm on :) 

Thanks for the review!